src/include/columnar/columnar.h

```diff
@@ -220,6 +220,10 @@ typedef struct TableReadState
 	TupleDesc tupleDescriptor;
 	Relation relation;
 
+	/* following are used for tables with zero columns */
+	uint64 totalRowCount;
+	uint64 readRowCount;
+
```

Comment:
These variables seem somewhat redundant with ``stripeReadRowCount`` and ``stripeBuffers->rowCount``. Should we move the special case of zero columns into the stripe-reading logic rather than the table-reading logic? I would prefer not have too many extra variables floating around that could become inconsistent with each other.

Reply:
Correct. But StripeBuffers loading logic depends on existence of `columnar.chunk` metadata for the table, which doesn't exist in the zero column case. I add a `columnar.chunk_group` metadata table in https://github.com/citusdata/citus/pull/4659 so we can refactor this and do as your suggestion in the future release.